presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -136,7 +138,12 @@ public WriteMapping toWriteMapping(Type type)
 
     private static ColumnMapping jsonColumnMapping()
     {
-        return ColumnMapping.sliceMapping(JSON, (resultSet, columnIndex) -> jsonParse(utf8Slice(resultSet.getString(columnIndex))), jsonWriteFunction(), domain -> Domain.all(domain.getType()));
+        return ColumnMapping.sliceMapping(JSON, (resultSet, columnIndex) -> jsonParse(utf8Slice(resultSet.getString(columnIndex))), jsonWriteFunction(), jsonPushdown());
+    }
+
+    private static UnaryOperator<Domain> jsonPushdown()
+    {
+        return domain -> Domain.all(VARCHAR);
```

Comment:
replace `jsonPushdown()` method with `io.prestosql.plugin.jdbc.ColumnMapping#DISABLE_PUSHDOWN` (static-import it)

here you're changing domain's type. Why so?

Reply:
I see the reason now. Can you check if https://github.com/prestosql/presto/pull/238 is sufficient? I didn't test that change with your code.

Reply:
Your fix works for me. Thanks!